From fdfb65e3d12080ab862a1847f7d2127a9fd3c6b4 Mon Sep 17 00:00:00 2001
From: Vladimir Barinov <vladimir.barinov@cogentembedded.com>
Date: Mon, 9 Mar 2020 23:15:28 +0300
Subject: [PATCH] media: platform: soc_camera: disable mutex locking for
 ADV_DEBUG

This disabled v4l2_ioctl locking for G/S_REGISTER.
Hence the this allows to use G/S_REGISTER and Q/DQBUG under
stress load.

Signed-off-by: Vladimir Barinov <vladimir.barinov@cogentembedded.com>
Signed-off-by: Nikita Yushchenko <nikita.yoush@cogentembedded.com>
---
 drivers/media/platform/soc_camera/soc_camera.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/drivers/media/platform/soc_camera/soc_camera.c b/drivers/media/platform/soc_camera/soc_camera.c
index 8084683..654b0df 100644
--- a/drivers/media/platform/soc_camera/soc_camera.c
+++ b/drivers/media/platform/soc_camera/soc_camera.c
@@ -2153,6 +2153,10 @@ static int soc_camera_video_start(struct soc_camera_device *icd)
 		v4l2_disable_ioctl(icd->vdev, VIDIOC_S_STD);
 		v4l2_disable_ioctl(icd->vdev, VIDIOC_ENUMSTD);
 	}
+#ifdef CONFIG_VIDEO_ADV_DEBUG
+	v4l2_disable_ioctl_locking(icd->vdev, VIDIOC_DBG_G_REGISTER);
+	v4l2_disable_ioctl_locking(icd->vdev, VIDIOC_DBG_S_REGISTER);
+#endif
 	ret = video_register_device(icd->vdev, VFL_TYPE_GRABBER, -1);
 	if (ret < 0) {
 		dev_err(icd->pdev, "video_register_device failed: %d\n", ret);
-- 
2.7.4

